chainerx_cc/chainerx/routines/linalg.cc

```diff
@@ -25,17 +27,33 @@ Array Dot(const Array& a, const Array& b) {
     // TODO(beam2d): dtype conversion
     CheckEqual(a.dtype(), b.dtype());
 
-    // TODO(beam2d): Support it. Need to transpose b so that the inner-product axis is moved to the top.
-    if (b.ndim() > 2) {
-        throw NotImplementedError{"dot does not support rhs operand with ndim > 2"};
-    }
+    Array modified_b;
```

Comment:
Could you be explicit with the instantiation? Sorry this is a style that's not yet documented so I wrote a temporary comment https://github.com/chainer/chainer/issues/5778#issuecomment-471480542
```suggestion
    Array modified_b{};
```